### ISSUE TITLE
Allow providing the `this` context to getValue and getKeyValue

### ISSUE THREAD

**Issue by matthewp** [canjs/can-reflect#8]
Reflect.get  takes a 3rd argument, the `this` value that will be passed into getters. This is useful. Would be nice to have in the can-reflect APIs as well.

**Comment by justinbmeyer** [comment]
What is this used for?

**Comment by matthewp** [comment]
Changing the context in getters.

**Comment by justinbmeyer** [comment]
Yeah, but why? I'm not arguing against changing it, I'm just wondering what is the use case. I understand why this is important for functions, but normally property access is so "bound/connected" to the object, in having trouble seeing where this is useful. Perhaps being able to "steal" something like Array's length.

**Comment by matthewp** [comment]
I don't have a specific use-case in can-reflect at the moment.  In regular old js, when overriding a Proxy getter you would want this feature so that if calling into the target getter the Proxy object would be the value of `this` (so if any property access happens there it also goes through the Proxy).

Just opened this for discussion, no reason to add until a clear use-case is present.

**Comment by matthewp** [comment]
Here's some pseudocode where this would be useful with Proxys. Might help think of a where this could also be important with can-reflect:

```js
var target = {
  foo: "bar"
};
Object.defineProperty(target, "baz", {
  get: function(){
    return this.foo;
  }
});

var p = new Proxy(target, {
  get: function(target, prop){
    // Count all observations
    Observation.add(target, prop);

    // This means that the `foo` accessor will be observed
    return Reflect.get(target, prop, p);
  }
});
```
